### PR TITLE
Issue 6 7

### PR DESCRIPTION
**Pull request by ros3bud**
This update separates out all of the alternates and makes it so that they will only show up as response variables when present. 

@cgrayson, LGTY?

### CODE REVIEW

File: spec/email_lookup_spec.coffee [hunk 1]
```diff
@@ -81,15 +81,51 @@ describe 'Email lookup', ->
...
+      assert.equal expected.email_lookup.alternate_email_1, response.email_lookup.alternate_email_1
+      assert.equal expected.email_lookup.alternate_email_2, response.email_lookup.alternate_email_2
+      assert.equal expected.email_lookup.alternate_email_3, response.email_lookup.alternate_email_3
+      assert.equal expected.email_lookup.alternate_email_4, response.email_lookup.alternate_email_4
+      assert.equal expected.email_lookup.alternate_email_5, response.email_lookup.alternate_email_5
 
-    it 'should return null when there is no alternate emails array', ->
-      nullAlternates = 
+    it 'should return the number of alternates present', ->
+      nullAlternates1 =
```
Comment (cgrayson) [inline spec/email_lookup_spec.coffee]: Is the name `nullAlternates1` right in this test (that does have alternates)?

File: spec/email_lookup_spec.coffee [hunk 2]
```diff
@@ -81,15 +81,51 @@ describe 'Email lookup', ->
...
+      expected =
+        email_lookup:
+          alternate_email_1: 'test@collegeclub.org'
+          alternate_email_2: 'test@il.rr.com'
+      response = emailIntegration.response({}, {}, nullAlternates1)
+      assert.equal expected.email_lookup.alternate_email_1, response.email_lookup.alternate_email_1
+      assert.equal expected.email_lookup.alternate_email_2, response.email_lookup.alternate_email_2
+
+    it 'should not return alternate emails', ->
+      nullAlternates2 =
```
Comment (cgrayson) [inline spec/email_lookup_spec.coffee]: I don't think a `2` is needed on this.

File: spec/email_lookup_spec.coffee [hunk 3]
```diff
@@ -110,9 +146,17 @@ describe 'Email lookup', ->
...
-          alternate_emails: null
-      response = emailIntegration.response({}, {}, nullAlternates)
-      assert.equal expected.email_lookup.alternate_emails, response.email_lookup.alternate_emails
+          alternate_email_1: undefined
+          alternate_email_2: undefined
+          alternate_email_3: undefined
+          alternate_email_4: undefined
+          alternate_email_5: undefined
+      response = emailIntegration.response({}, {}, nullAlternates2)
+      assert.equal expected.email_lookup_email_1, response.email_lookup.alternate_email_1
```
Comment (cgrayson) [inline spec/email_lookup_spec.coffee]: I think you could use `assert.isUndefined()` for these.

File: spec/email_lookup_spec.coffee [hunk 4]
```diff
@@ -81,15 +81,51 @@ describe 'Email lookup', ->
...
         email_lookup:
-          alternate_emails: 'test@collegeclub.org (09/15/2008),test@il.rr.com (date unavailable),test@collegeclub.com (02/21/2012),test@excite.com (date unavailable)'
+          alternate_email_1: 'test@collegeclub.org'
+          alternate_email_2: 'test@il.rr.com'
+          alternate_email_3: 'test@collegeclub.com'
+          alternate_email_4: 'test@excite.com'
+          alternate_email_5: 'tester@test.com'
       response = emailIntegration.response({}, {}, @res)
-      assert.equal expected.email_lookup.alternate_emails, response.email_lookup.alternate_emails
+      assert.equal expected.email_lookup.alternate_email_1, response.email_lookup.alternate_email_1
```
Comment (cgrayson) [inline spec/email_lookup_spec.coffee]: Oops, I didn't notice this previously, but the order of these parameters is wrong. It should be `assert.equal(actual, expected)`.

File: spec/phone_lookup_spec.coffee [hunk 5]
```diff
@@ -88,38 +88,79 @@ describe 'Phone lookup', ->
...
+          alternate_phone_2: '5555555555'
+          alternate_phone_3: '5125551212'
+          alternate_phone_4: '5125554343'
+          alternate_phone_5: '5125551414'
       response = phoneIntegration.response({}, {}, @successRes)
-      assert.equal expected.phone_lookup.alternate_phones, response.phone_lookup.alternate_phones
-
-    it 'should return null when there are no alternate phones', ->
-      alternatePhonesRes =
+      assert.equal expected.phone_lookup.alternate_phone_1, response.phone_lookup.alternate_phone_1
```
Comment (cgrayson) [inline spec/phone_lookup_spec.coffee]: Come to think of it, I wonder if `assert.deepEqual()` could be used to reduce some of this repetition.

File: src/common.coffee [hunk 6]
```diff
@@ -96,9 +103,8 @@ parseAlternatePhones = (value) ->
   newArray = []
   for element, i in valueArray
     altPhone = valueArray[i].match(/\d{10}/)
-    altPhoneDate = parseTimestamp(valueArray[i].match(/\((.*)\)/)[1])
-    newArray[i] = "#{altPhone} (#{altPhoneDate})"
-  newArray.join()
+    newArray[i] = "#{altPhone}"
```
Comment (cgrayson) [inline src/common.coffee]: I wonder if the variable `altPhone` is needed? Could the `match` above be assigned right to `newArray[i]`?

File: spec/email_lookup_spec.coffee (hunk 1 above)
Comment (ros3bud) [inline spec/email_lookup_spec.coffee]: Nope - it's not correct any longer. Thanks for catching.

File: spec/email_lookup_spec.coffee (hunk 2 above)
Comment (ros3bud) [inline spec/email_lookup_spec.coffee]: correct - similar situation. Thanks.

**Comment by cgrayson** [comment]
As discussed offline, look into changing to store the `alternate` emails & phones in an array. Let me know if any other questions come up from that, or when this is ready to look over again.

**Comment by ros3bud** [comment]
Sounds good. I created a new branch for the alternates arrays.

File: src/common.coffee (hunk 6 above)
Comment (ros3bud) [inline src/common.coffee]: It works with a direct assignment - thanks!

File: spec/phone_lookup_spec.coffee (hunk 5 above)
Comment (ros3bud) [inline spec/phone_lookup_spec.coffee]: now utilizing deepEqual to check the alternate arrays.

File: spec/email_lookup_spec.coffee (hunk 4 above)
Comment (ros3bud) [inline spec/email_lookup_spec.coffee]: Updated in both tests.

File: spec/email_lookup_spec.coffee (hunk 3 above)
Comment (ros3bud) [inline spec/email_lookup_spec.coffee]: True, but no longer doing this.

**Comment by ros3bud** [comment]
Strategy changed here so closing this up. Alternate arrays are created instead of this approach.

**Pull request closed without merging**